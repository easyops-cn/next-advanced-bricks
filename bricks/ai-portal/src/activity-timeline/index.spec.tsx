import { describe, test, expect, jest } from "@jest/globals";
import { act } from "react-dom/test-utils";
import "./";
import type { ActivityTimeline } from "./index.js";
import type { Activity } from "./interfaces";

jest.mock("@next-core/theme", () => ({}));

beforeAll(() => {
  jest.useFakeTimers().setSystemTime(new Date("2025-09-18"));
});

afterAll(() => {
  jest.useRealTimers();
});

describe("ai-portal.activity-timeline", () => {
  test("basic usage", async () => {
    const element = document.createElement(
      "ai-portal.activity-timeline"
    ) as ActivityTimeline;
    element.list = [
      {
        user_id: "u001",
        user_name: "Tom",
        action_type: "create_goal",
        time: 1757853597000,
      },
      {
        user_id: "u001",
        user_name: "Tom",
        action_type: "alter_owner",
        time: 1757863597000,
        metadata: {
          after: {
            owner: {
              user_name: "Lucy",
            },
          },
        },
      },
      {
        user_id: "u002",
        user_name: "Lucy",
        action_type: "start_conversation",
        time: 1757904096000,
        metadata: {
          conversation_id: "c001",
          conversation_title: "项目规划",
        },
      },
      {
        user_id: "u002",
        user_name: "Lucy",
        action_type: "decompose_goals",
        time: 1757904096000,
        metadata: {
          sub_goals_count: 2,
          sub_goals: [
            {
              title: "先计划",
            },
            {
              title: "再执行",
            },
          ],
        },
      },
      {
        user_id: "u002",
        user_name: "Lucy",
        action_type: "alter_user",
        time: 1757904096000,
        metadata: {
          before: {
            users: [
              {
                user_name: "Jim",
              },
            ],
          },
          after: {
            users: [
              {
                user_name: "Joy",
              },
              {
                user_name: "Green",
              },
            ],
          },
        },
      },
      {
        user_id: "u001",
        user_name: "Tom",
        action_type: "add_comment",
        time: 1757904096000,
        metadata: {
          comment_content: "Good!",
        },
      },
    ] as Activity[];

    act(() => {
      document.body.appendChild(element);
    });

    expect(element.shadowRoot?.childNodes).toMatchSnapshot();

    act(() => {
      document.body.removeChild(element);
    });
  });

  test("loading state", async () => {
    const element = document.createElement(
      "ai-portal.activity-timeline"
    ) as ActivityTimeline;

    act(() => {
      document.body.appendChild(element);
    });
    expect(element.shadowRoot?.querySelector(".loading")).toBeTruthy();

    act(() => {
      document.body.removeChild(element);
    });
  });
});
