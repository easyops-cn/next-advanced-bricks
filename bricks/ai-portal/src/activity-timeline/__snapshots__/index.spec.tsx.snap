// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ai-portal.activity-timeline basic usage 1`] = `
NodeList [
  <style>
    styles.shadow.css
  </style>,
  <ul>
    <li
      class="activity"
    >
      <div
        class="guide-line"
      />
      <div
        class="main"
      >
        <eo-easyops-avatar
          class="avatar"
          nameorinstanceid="u001"
          size="xs"
        />
        <span
          class="user"
        >
          Tom
        </span>
         
        <span
          class="action"
        >
          created this goal
        </span>
        <span
          class="time"
          title="September 14, 2025 20:39"
        >
          3 days ago
        </span>
      </div>
    </li>
    <li
      class="activity"
    >
      <div
        class="guide-line"
      />
      <div
        class="main"
      >
        <eo-easyops-avatar
          class="avatar"
          nameorinstanceid="u001"
          size="xs"
        />
        <span
          class="user"
        >
          Tom
        </span>
         
        <span
          class="action"
        >
          set Lucy as the owner
        </span>
        <span
          class="time"
          title="September 14, 2025 23:26"
        >
          3 days ago
        </span>
      </div>
    </li>
    <li
      class="activity"
    >
      <div
        class="guide-line"
      />
      <div
        class="main"
      >
        <eo-easyops-avatar
          class="avatar"
          nameorinstanceid="u002"
          size="xs"
        />
        <span
          class="user"
        >
          Lucy
        </span>
         
        <span
          class="action"
        >
          <span>
            started a chat
          </span>
          <eo-link
            class="conversation"
            target="_blank"
          >
            项目规划
          </eo-link>
          <eo-icon
            class="external-link-icon"
            icon="external-link"
            lib="lucide"
          />
        </span>
        <span
          class="time"
          title="September 15, 2025 10:41"
        >
          3 days ago
        </span>
      </div>
    </li>
    <li
      class="activity"
    >
      <div
        class="guide-line"
      />
      <div
        class="main"
      >
        <eo-easyops-avatar
          class="avatar"
          nameorinstanceid="u002"
          size="xs"
        />
        <span
          class="user"
        >
          Lucy
        </span>
         
        <span
          class="action"
        >
          decomposed this goal to 2 sub-goals: 先计划, 再执行
        </span>
        <span
          class="time"
          title="September 15, 2025 10:41"
        >
          3 days ago
        </span>
      </div>
    </li>
    <li
      class="activity"
    >
      <div
        class="guide-line"
      />
      <div
        class="main"
      >
        <eo-easyops-avatar
          class="avatar"
          nameorinstanceid="u002"
          size="xs"
        />
        <span
          class="user"
        >
          Lucy
        </span>
         
        <span
          class="action"
        >
          removed a goal member: Jim; added 2 goal members: Joy, Green
        </span>
        <span
          class="time"
          title="September 15, 2025 10:41"
        >
          3 days ago
        </span>
      </div>
    </li>
    <li
      class="activity"
    >
      <div
        class="guide-line"
      />
      <div
        class="main"
      >
        <eo-easyops-avatar
          class="avatar"
          nameorinstanceid="u001"
          size="xs"
        />
        <span
          class="user"
        >
          Tom
        </span>
         
        <span
          class="action"
        >
          commented
        </span>
        <span
          class="time"
          title="September 15, 2025 10:41"
        >
          3 days ago
        </span>
      </div>
      <div
        class="comment"
      >
        Good!
      </div>
    </li>
  </ul>,
]
`;
